config/v1/types_infrastructure.go

```diff
@@ -172,6 +172,9 @@ const (
 
 	// NutanixPlatformType represents Nutanix infrastructure.
 	NutanixPlatformType PlatformType = "Nutanix"
+
+	// ExternalPlatformType represent generic infrastructure provider. Provider-specific components should be supplemented separately.
```

Comment:
i'm a little hesitant about using the word "provider" so much, my personal preference is to focus more on using the word "infrastructure" to define the underlying platforms we are talking about. i might say `Infrastructure-specific components ...`, or maybe we should stick with the "platform" language used by the API.

Reply:
In regard to "platform" - might this be confused with things like `x86` or `arm`?

I do like `Infrastructure-specific components ` wording, will update the patch.

Reply:
I had another look at the code, and have a feeling that using "platform" language will be better, for the sake of consistency.

Reply:
> In regard to "platform" - might this be confused with things like x86 or arm? 

it's possible, good call out. i would hope we are differentiating those by using the term "architecture", but i hope the context here would help.